src/Illuminate/Routing/Matching/SubdomainValidator.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Illuminate\Routing\Matching;
+
+use Illuminate\Http\Request;
+use Illuminate\Routing\Route;
+
+class SubdomainValidator implements ValidatorInterface
+{
+    /**
+     * Validate a given rule against a route and request.
+     *
+     * @param  \Illuminate\Routing\Route  $route
+     * @param  \Illuminate\Http\Request  $request
+     * @return bool
+     */
+    public function matches(Route $route, Request $request)
+    {
+        if (is_null($route->subdomain())) {
+            return true;
+        }
+
+        $subdomains = implode('|', $route->subdomain());
+        $regex = "/^$subdomains\..*/i";
```

Comment:
I agree that we don't want to make it unnecessarily complex. 

I agree that the parentheses "can be" optional, but only if you aren't adding additional chars after them. But, you are.

So, your `.` is still not actually doing what you intended.

Your tests are "accidentally" working, because the `.*` (wildcard) also accepts a `.`

In the example: 

```php
$subdomains = 'api|backend';
$regex = "/^$subdomains\..*/i";
```

this translates to:

```php
preg_match("/^api|backend\..*/i", foo)
```

Notice there that the `|` means that the preg_match is going to process the following as possible values:

`api.*`
`backend\..*`

Notice the inconsistency. 

Options:
a) use parentheses to be specific
b) in your `implode`, use `\.|` instead of just `|` to be explicit about wanting a trailing `.` at the end of each subdomain.

It may seem small, but this is the router we're talking about. Accuracy is important.